### ISSUE TITLE
Use the latest foundry-config and foundry-compilers

### ISSUE THREAD

**Issue by TilakMaddy** [Cyfrin/aderyn#526]
Run `aderyn` on this first flight

https://github.com/Cyfrin/2024-06-Dussehra

```
thread 'main' panicked at /Users/tilakmadichetti/.cargo/registry/src/index.crates.io-6f17d22bba15001f/foundry-config-0.2.0/src/lib.rs:466:55:
failed to extract foundry config:
foundry config error: Unknown evm version: cancun for setting `evm_version`

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by TilakMaddy** [comment]
@alexroan Can we bring back the foundry config as it was in ICF branch (where the latest foundry repo was a git submodule dependency)? You have explained some problems with it while publishing the package (Example - lack of version, etc) Can we set the version ourselves and publish? I don't think it's a good idea to rely on this 8 month old crate that never got any updates

**Comment by TilakMaddy** [comment]
💡 IDEA

Create a fork of the latest foundry codebase, and publish foundry-config crate under a different name like cyfrin-foundry-config and we can use that : )

**Comment by alexroan** [comment]
I'll try to get this done (similar to how we approached foundry-compilers), but there may be reasons why they stopped publishing.

**Comment by chista0x** [comment]
I have this problem too

**Comment by alexroan** [comment]
@TilakMaddy It looks as though Foundry is releasing up-to-date versions of foundry-compilers and foundry-config again, so we can switch to using that instead of forking and publishing our own.

However, this looks like it requires some rewriting and dependency management of ICF.

I'm looking into this now.

**Comment by TilakMaddy** [comment]
Fix in #564
